### PR TITLE
INGITE-9285 Add MaxAbs scaler

### PR DESCRIPTION
**Pull request by dehasi**

### CODE REVIEW

**Comment by zaleslaw** [comment]
I'll review this ticket in a few days

File: examples/src/main/java/org/apache/ignite/examples/ml/preprocessing/MaxAbsScalerExample.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+ * in this example. The second preprocessor here is a MinMaxScaler preprocessor which is built on top of the feature
+ * extractor and represents a chain of itself and the underlying feature extractor.
+ */
+public class MaxAbsScalerExample {
+    /** Run example. */
+    public static void main(String[] args) throws Exception {
+        try (Ignite ignite = Ignition.start("examples/config/example-ignite.xml")) {
+            System.out.println(">>> Max abs example started.");
+
+            IgniteCache<Integer, Person> persons = createCache(ignite);
```
Comment (zaleslaw) [inline examples/src/main/java/org/apache/ignite/examples/ml/preprocessing/MaxAbsScalerExample.java]: Please update this example according last changes in the master (look at the another preprocessors)

File: modules/ml/src/main/java/org/apache/ignite/ml/preprocessing/maxabsscaling/MaxAbsScalerTrainer.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import org.apache.ignite.ml.dataset.Dataset;
+import org.apache.ignite.ml.dataset.DatasetBuilder;
+import org.apache.ignite.ml.dataset.UpstreamEntry;
+import org.apache.ignite.ml.dataset.primitive.context.EmptyContext;
+import org.apache.ignite.ml.math.functions.IgniteBiFunction;
+import org.apache.ignite.ml.math.primitives.vector.Vector;
+import org.apache.ignite.ml.preprocessing.PreprocessingTrainer;
+
+/**
+ * Trainer of the maxabsscaling preprocessor.
```
Comment (zaleslaw) [inline modules/ml/src/main/java/org/apache/ignite/ml/preprocessing/maxabsscaling/MaxAbsScalerTrainer.java]: Change this comment to more readable format

File: examples/src/main/java/org/apache/ignite/examples/ml/preprocessing/MaxAbsScalerExample.java (hunk 1 above)
Comment (dehasi) [reply examples/src/main/java/org/apache/ignite/examples/ml/preprocessing/MaxAbsScalerExample.java]: I updated this example according to `MinMaxScalerExample`.

File: modules/ml/src/main/java/org/apache/ignite/ml/preprocessing/maxabsscaling/MaxAbsScalerTrainer.java (hunk 2 above)
Comment (dehasi) [reply modules/ml/src/main/java/org/apache/ignite/ml/preprocessing/maxabsscaling/MaxAbsScalerTrainer.java]: The most `*Trainer` classes have such java-doc template `Trainer of the <TrainerName> preprocessor`. So did I.

**Pull request closed without merging**